File: src/main/scala/fi/oph/koski/config/ConfigForFrontend.scala
package fi.oph.koski.config

import com.typesafe.config.Config

import java.time.LocalDate
import java.time.format.DateTimeFormatter

case class ConfigForFrontend(
  rajapäivät: Rajapäivät,
)

object ConfigForFrontend {
  def apply(config: Config): ConfigForFrontend = ConfigForFrontend(
    rajapäivät = Rajapäivät(config),
  )
}

case class Rajapäivät(
  ibLaajuusOpintopisteinäAlkaen: String,
)

object Rajapäivät {
  def apply(config: Config): Rajapäivät = Rajapäivät(
    ibLaajuusOpintopisteinäAlkaen = dateString(config, "validaatiot.ibLaajuudetOpintopisteinäAlkaen"),
  )

  // Tarkastaa että päivämäärä on validi ja varmistaa sen formaatin frontin ymmärtämässä muodossa
  def dateString(config: Config, key: String): String =
    LocalDate.parse(config.getString(key)).format(DateTimeFormatter.ISO_LOCAL_DATE)
}

File: web/app/types/fi/oph/koski/config/ConfigForFrontend.ts
import { Rajapäivät } from './Rajapaivat'

/**
 * ConfigForFrontend
 *
 * @see `fi.oph.koski.config.ConfigForFrontend`
 */
export type ConfigForFrontend = {
  $class: 'fi.oph.koski.config.ConfigForFrontend'
  rajapäivät: Rajapäivät
}

export const ConfigForFrontend = (o: {
  rajapäivät: Rajapäivät
}): ConfigForFrontend => ({
  $class: 'fi.oph.koski.config.ConfigForFrontend',
  ...o
})

ConfigForFrontend.className = 'fi.oph.koski.config.ConfigForFrontend' as const

export const isConfigForFrontend = (a: any): a is ConfigForFrontend =>
  a?.$class === 'fi.oph.koski.config.ConfigForFrontend'
